### ISSUE TITLE
[BUG] iloc is very slow

### ISSUE THREAD

**Issue by mrocklin** [rapidsai/cudf#1176]
```python
In [1]: import cudf, dask.dataframe as dd, numpy as np

In [2]: df = cudf.DataFrame({'x': np.arange(100000)})

In [3]: %time df.iloc[1000:100000]
CPU times: user 10.3 s, sys: 136 ms, total: 10.5 s
Wall time: 10.5 s
Out[3]: <cudf.DataFrame ncols=1 nrows=99000 >
```

**Comment by kkraus14** [comment]
This will not get fixed until 0.7.

**Comment by mrocklin** [comment]
This hurts us in joins as well.  I'm inclined to add a special case to `iloc` that passes things through if the slice is just the entire thing.  Would this go in 0.6?

**Comment by kkraus14** [comment]
> This hurts us in joins as well. I'm inclined to add a special case to `iloc` that passes things through if the slice is just the entire thing. Would this go in 0.6?

Let me see if I can just rip out the entire iloc implementation currently and pass it through to __getitem__ for now.

**Comment by mrocklin** [comment]
This was my local hack

```diff
diff --git a/python/cudf/dataframe/dataframe.py b/python/cudf/dataframe/dataframe.py
index 9a8e155..f8a8419 100644
--- a/python/cudf/dataframe/dataframe.py
+++ b/python/cudf/dataframe/dataframe.py
@@ -2471,6 +2471,9 @@ class Iloc(object):
             rows.append(arg)

         elif isinstance(arg, slice):
+            if arg == slice(None, None) or arg == slice(0, len(self._df)):
+                return self._df
+
             start, stop, step, sln = utils.standard_python_slice(len_idx, arg)
             if sln > 0:
                 for idx in range(start, stop, step):
diff --git a/python/cudf/dataframe/series.py b/python/cudf/dataframe/series.py
index 637f03e..caaa0f8 100644
--- a/python/cudf/dataframe/series.py
+++ b/python/cudf/dataframe/series.py
@@ -1465,6 +1465,9 @@ class Iloc(object):
             rows.append(arg)

         elif isinstance(arg, slice):
+            if arg == slice(None, None) or arg == slice(0, len(self._df)):
+                return self._df
+
             start, stop, step, sln = utils.standard_python_slice(len_idx, arg)
             if sln > 0:
                 for idx in range(start, stop, step):
```

**Comment by kkraus14** [comment]
> This was my local hack
> 
> ```diff
> diff --git a/python/cudf/dataframe/dataframe.py b/python/cudf/dataframe/dataframe.py
> index 9a8e155..f8a8419 100644
> --- a/python/cudf/dataframe/dataframe.py
> +++ b/python/cudf/dataframe/dataframe.py
> @@ -2471,6 +2471,9 @@ class Iloc(object):
>              rows.append(arg)
> 
>          elif isinstance(arg, slice):
> +            if arg == slice(None, None) or arg == slice(0, len(self._df)):
> +                return self._df
> +
>              start, stop, step, sln = utils.standard_python_slice(len_idx, arg)
>              if sln > 0:
>                  for idx in range(start, stop, step):
> diff --git a/python/cudf/dataframe/series.py b/python/cudf/dataframe/series.py
> index 637f03e..caaa0f8 100644
> --- a/python/cudf/dataframe/series.py
> +++ b/python/cudf/dataframe/series.py
> @@ -1465,6 +1465,9 @@ class Iloc(object):
>              rows.append(arg)
> 
>          elif isinstance(arg, slice):
> +            if arg == slice(None, None) or arg == slice(0, len(self._df)):
> +                return self._df
> +
>              start, stop, step, sln = utils.standard_python_slice(len_idx, arg)
>              if sln > 0:
>                  for idx in range(start, stop, step):
> ```

I'm going to fix it generally, I don't think it's super high effort.

**Comment by kkraus14** [comment]
Fixed in #1184
